org.eclipse.lsp4e/src/org/eclipse/lsp4e/Versioned.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2022-3 Cocotec Ltd and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Ahmed Hussain (Cocotec Ltd) - initial implementation
+ *
+ *******************************************************************************/
+package org.eclipse.lsp4e;
+
+/**
+ * Bundles together a result from a language server with the document version at the time it was run.
+ * Supports optimistic locking.
+ *
+ * @param <T>
+ */
+public class Versioned<T> {
+	private final long version;
+
+	private final T data;
+
+	public Versioned(final long version, final T data) {
```

Comment:
what version should that be here? The documentVersion from the Eclipse document? Please clarify that in name and javadoc.